client/login/redirect-logged-in/index.web.js

```diff
@@ -1,14 +1,38 @@
 import safeProtocolUrl from 'calypso/lib/safe-protocol-url';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 
+/**
+ * For this context, we consider external URLs that are NOT:
+ * - Relative paths (`/test`)
+ * - Absolute URLs on https://wordpress.com/*
+ * @param {string} url URL to check
+ * @returns {boolean}
+ */
+function isExternalUrl( url ) {
+	if ( url.startsWith( '/' ) ) {
+		return false;
+	}
+
+	try {
+		const urlObject = new URL( url );
+		if ( urlObject.hostname === 'wordpress.com' && urlObject.protocol === 'https:' ) {
```

Comment:
Since we now check the protocol here, should we remove the `safeProtocolUrl` call?

Reply:
Good call. Cleaned up in f16484537aa1b709996742a78eff3bfa8b070bba